src/RadixDlt.NetworkGateway.PostgresIntegration/Services/EntityStateQuerier.cs

```diff
@@ -170,15 +172,23 @@ public async Task<GatewayModel.StateEntityDetailsResponse> EntityDetails(List<Gl
             switch (entity)
             {
                 case FungibleResourceEntity frme:
+                    var fungibleResourceSupplyData = resourcesSupplyData!.Single(x => x.ResourceEntityId == frme.Id);
```

Comment:
null-negation stinks... guess we might want to assign "empty dictionary" rather null then?